src/realm/object-store/sync/sync_session.cpp

```diff
@@ -237,21 +237,19 @@ void SyncSession::become_waiting_for_access_token()
     m_state = State::WaitingForAccessToken;
 }
 
-void SyncSession::handle_bad_auth(const std::shared_ptr<SyncUser>& user, Status error_code,
-                                  std::string_view context_message)
+void SyncSession::handle_bad_auth(const std::shared_ptr<SyncUser>& user, Status status)
 {
     // TODO: ideally this would write to the logs as well in case users didn't set up their error handler.
     {
         util::CheckedUniqueLock lock(m_state_mutex);
-        cancel_pending_waits(std::move(lock), error_code);
+        cancel_pending_waits(std::move(lock), status);
     }
     if (user) {
         user->log_out();
     }
 
     if (auto error_handler = config(&SyncConfig::error_handler)) {
-        auto user_facing_error =
-            SyncError(Status{realm::sync::ProtocolError::bad_authentication, context_message}, true);
+        auto user_facing_error = SyncError({ErrorCodes::AuthError, status.reason()}, true);
```

Comment:
you need to set the correct action (LogOutUser).

Reply:
So my confusion here is that this is not getting passed to `handle_error()`, it's getting passed to `error_handler` and the action actually has no effect. I can set it here for completeness, but it's not going to get used.

Reply:
ah, I see. It's calling the user callback directly.

Reply:
is it the right thing though? 🤔

Reply:
Not sure what you mean? What else would we do here?

Reply:
so I was wondering if we should call  `handle_error()` instead, so we actually log out the user.

Reply:
But we log out the user right above on line 248

Reply:
The session should already be closed at this point: this path is called when a SyncSession is being revived and will never `become_active()` on a bad auth or if the websocket connection failed for bad auth or redirect during open.